### PR TITLE
Color pickers added to options tab to allow users choose band colors

### PR DESCRIPTION
**Pull request by pacurtin**

### CODE REVIEW

File: public/prelert_swimlane_vis_controller.js [hunk 1]
```diff
@@ -263,7 +263,14 @@ module.controller('PrelertSwimlaneVisController', function ($scope, courier) {
 
       // Create a series for each severity color band,
       // plus an 'unknown' series for scores less than the 'low' threshold.
-      const colorBands = ['#e6e6e6', '#d2e9f7', '#8bc8fb', '#ffdd00', '#ff7e00', '#fe5050'];
+      //If color pickers have been used use new color choice else go to default colors.
+      const colorBands = [scope.vis.params.unknownThresholdColor ? scope.vis.params.unknownThresholdColor :'#e6e6e6',
```
Comment (peteharverson) [inline public/prelert_swimlane_vis_controller.js]: Could you define the default colors in `prelert_swimlane_vis.js`, in the `params.defaults` block. That way the inputs are filled out with the default hexadecimal colors on opening. i.e.
```
params : {
      editor : require('plugins/prelert_swimlane_vis/prelert_swimlane_vis_params.html'),
      defaults : {
        interval : { display : 'Auto', val : 'auto' },
        lowThreshold: 0,
        warningThreshold: 3,
        minorThreshold: 25,
        majorThreshold: 50,
        criticalThreshold: 75,
        lowThresholdColor : '#d2e9f7',
        warningThresholdColor : '#8bc8fb',
        minorThresholdColor : '#ffdd00',
        majorThresholdColor : '#ff7e00',
        criticalThresholdColor : '#fe5050',
        tooltipNumberFormat: '0.0',
        showLegend: true
      },
      intervalOptions...
```

File: public/prelert_swimlane_vis_params.html [hunk 2]
```diff
@@ -1,53 +1,63 @@
...
 
   <div class="threshold-input-container">
-    <i class="fa fa-circle icon-severity-low" aria-hidden="true"></i>
+    <i class="fa fa-circle icon-severity-low" ng-style="{'color':vis.params.lowThresholdColor}" aria-hidden="true"></i>
     <input type="number" name="low_threshold_input" step="any" class="form-control"
       ng-model="vis.params.lowThreshold" ng-model-options="{ updateOn: 'blur' }"
       less-than="{{vis.params.warningThreshold}}"
       ng-required="true">
-    <kbn-info info="Defines lower threshold for values between {{vis.params.lowThreshold}} and {{vis.params.warningThreshold}}"></kbn-info>
+    <kbn-info info="Defines lower threshold for values between {{vis.params.lowThreshold}} and {{vis.params.warningThreshold}}."></kbn-info>
```
Comment (peteharverson) [inline public/prelert_swimlane_vis_params.html]: No need for the full stop in the tooltip for the first field since it is only one sentence.

**Review by peteharverson — COMMENTED** [review]
Many thanks for adding this functionality! Being able to configure custom colors for the bands is a great addition. If you could just make the minor edit to the tooltip in the low threshold input, and define the default colors in prelert_swimlane_vis.js, then this can be merged in and I will produce a new release.

**Comment by pacurtin** [comment]
Ok. I've refactored it to look at defaults for the colorpicker fields and changed that tooltip typo. It looks like it works but maybe give it a quick test before you merge it. Let me know if there is anything else you would like changed.

**Review by peteharverson — APPROVED** [review]
LGTM. Thanks for making the edits. I have tested it and all working well. I will merge it, and then package up a new 5.3.2 release.

**Pull request merged**